File: scripts/generate_metadata.py
import os
import re
import json

root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

# Helper to determine category from directory name
def get_category(dir_name):
    name = dir_name.lower()
    if 'onboarding' in name or 'tutorial' in name:
        return 'Onboarding'
    if 'installation' in name or 'setup_implementation' in name:
        return 'Installation & Setup'
    if any(k in name for k in ['mfa', 'totp', 'verification', 'audit_logs', 'security_audit', 'authgate']):
        return 'Security & MFA'
    if any(k in name for k in ['checkout', 'payout', 'billing', 'subscription', 'affiliate', 'revenue', 'payment']):
        return 'Billing & Affiliates'
    if any(k in name for k in ['license', 'ea_marketplace', 'secure_hub', 'gating']):
        return 'Licensing & Security Hub'
    if any(k in name for k in ['desktop', 'elite_dashboard', 'pro_terminal', 'execution_terminal', 'live_terminal']):
        return 'Desktop Terminals'
    if any(k in name for k in ['mobile', 'hq_dashboard', 'android_simulator', 'distribution_hub', 'play_store_focus', 'store_listing']):
        return 'Mobile HQ & Simulators'
    if 'bridge' in name or 'sync' in name:
        return 'MT4/MT5 Bridge Sync'
    if any(k in name for k in ['neural', 'volatility', 'predictive', 'stress_test', 'backtester', 'correlation', 'sentiment', 'agent', 'automation', 'backtest']):
        return 'Neural Engines & Algos'
    if any(k in name for k in ['journal', 'analytics', 'performance', 'recap', 'audit']):
        return 'Trade Journal & Analytics'
    if any(k in name for k in ['chart', 'heatmap', 'liquidity', 'movers', 'feed', 'sentiment']):
        return 'Live Charts & Market Feeds'
    if 'consensus' in name:
        return 'Consensus Matrix'
    return 'General UI Modules'

# Helper to determine style theme based on code content
def get_theme(html_content):
    if 'Geist' in html_content or '#00ffa3' in html_content or '#00e290' in html_content:
        return {
            'name': 'Obsidian Aegis',
            'badgeClass': 'bg-emerald-500/10 text-emerald-400 border border-emerald-500/20'
        }
    if 'Space Grotesk' in html_content or '#0e141a' in html_content or '#dee3ec' in html_content:
        return {
            'name': 'Obsidian Abyss',
            'badgeClass': 'bg-blue-500/10 text-blue-400 border border-blue-500/20'
        }
    return {
        'name': 'Precision Sniper ICT',
        'badgeClass': 'bg-lime-500/10 text-lime-400 border border-lime-500/20'
    }

def main():
    print("Scanning workspace for code.html files...")
    screens = []

    for entry in os.listdir(root_dir):
        entry_path = os.path.join(root_dir, entry)
        if not os.path.isdir(entry_path):
            continue
        if entry in ['node_modules', '.git', 'backend', 'scripts', '.agents', 'shader', 'three.js']:
            continue

        code_html_path = os.path.join(entry_path, 'code.html')
        if not os.path.exists(code_html_path):
            continue

        try:
            with open(code_html_path, 'r', encoding='utf-8') as f:
                html_content = f.read()

            # Extract title
            title_match = re.search(r'<title>([^<]+)</title>', html_content, re.IGNORECASE)
            title = title_match.group(1).strip() if title_match else entry.replace('_', ' ').upper()

            # Check if screen.png exists
            screen_png_path = os.path.join(entry_path, 'screen.png')
            has_screenshot = os.path.exists(screen_png_path)

            theme = get_theme(html_content)
            category = get_category(entry)

            screens.append({
                'folderName': entry,
                'title': title,
                'path': f"{entry}/code.html",
                'screenshot': f"{entry}/screen.png" if has_screenshot else None,
                'category': category,
                'theme': theme
            })
        except Exception as e:
            print(f"Error reading {entry}: {e}")

    # Sort screens alphabetically by category then by title
    screens.sort(key=lambda s: (s['category'], s['title']))

    output_path = os.path.join(root_dir, 'screens.json')
    with open(output_path, 'w', encoding='utf-8') as f:
        json.dump(screens, f, indent=2)

    print(f"Successfully indexed {len(screens)} screens. Output saved to: {output_path}")

if __name__ == '__main__':
    main()

File: scripts/generate_metadata.js
import fs from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const rootDir = path.join(__dirname, '..');

// Helper to determine category from directory name
function getCategory(dirName) {
    const name = dirName.toLowerCase();
    
    if (name.includes('onboarding') || name.includes('tutorial')) {
        return 'Onboarding';
    }
    if (name.includes('installation') || name.includes('setup_implementation')) {
        return 'Installation & Setup';
    }
    if (name.includes('mfa') || name.includes('totp') || name.includes('verification') || name.includes('audit_logs') || name.includes('security_audit') || name.includes('authgate')) {
        return 'Security & MFA';
    }
    if (name.includes('checkout') || name.includes('payout') || name.includes('billing') || name.includes('subscription') || name.includes('affiliate') || name.includes('revenue') || name.includes('payment')) {
        return 'Billing & Affiliates';
    }
    if (name.includes('license') || name.includes('ea_marketplace') || name.includes('secure_hub') || name.includes('gating')) {
        return 'Licensing & Security Hub';
    }
    if (name.includes('desktop') || name.includes('elite_dashboard') || name.includes('pro_terminal') || name.includes('execution_terminal') || name.includes('live_terminal')) {
        return 'Desktop Terminals';
    }
    if (name.includes('mobile') || name.includes('hq_dashboard') || name.includes('android_simulator') || name.includes('distribution_hub') || name.includes('play_store_focus') || name.includes('store_listing')) {
        return 'Mobile HQ & Simulators';
    }
    if (name.includes('bridge') || name.includes('sync')) {
        return 'MT4/MT5 Bridge Sync';
    }
    if (name.includes('neural') || name.includes('volatility') || name.includes('predictive') || name.includes('stress_test') || name.includes('backtester') || name.includes('correlation') || name.includes('sentiment') || name.includes('agent') || name.includes('automation') || name.includes('backtest')) {
        return 'Neural Engines & Algos';
    }
    if (name.includes('journal') || name.includes('analytics') || name.includes('performance') || name.includes('recap') || name.includes('audit')) {
        return 'Trade Journal & Analytics';
    }
    if (name.includes('chart') || name.includes('heatmap') || name.includes('liquidity') || name.includes('movers') || name.includes('feed') || name.includes('sentiment')) {
        return 'Live Charts & Market Feeds';
    }
    if (name.includes('consensus')) {
        return 'Consensus Matrix';
    }
    return 'General UI Modules';
}

// Helper to determine style theme based on code content
function getTheme(htmlContent) {
    if (htmlContent.includes('Geist') || htmlContent.includes('#00ffa3') || htmlContent.includes('#00e290')) {
        return {
            name: 'Obsidian Aegis',
            badgeClass: 'bg-emerald-500/10 text-emerald-400 border border-emerald-500/20'
        };
    }
    if (htmlContent.includes('Space Grotesk') || htmlContent.includes('#0e141a') || htmlContent.includes('#dee3ec')) {
        return {
            name: 'Obsidian Abyss',
            badgeClass: 'bg-blue-500/10 text-blue-400 border border-blue-500/20'
        };
    }
    return {
        name: 'Precision Sniper ICT',
        badgeClass: 'bg-lime-500/10 text-lime-400 border border-lime-500/20'
    };
}

async function main() {
    console.log("Scanning workspace for code.html files...");
    const entries = await fs.readdir(rootDir, { withFileTypes: true });
    const screens = [];

    for (const entry of entries) {
        if (!entry.isDirectory()) continue;
        if (['node_modules', '.git', 'backend', 'scripts', '.agents', 'shader', 'three.js'].includes(entry.name)) continue;

        const folderPath = path.join(rootDir, entry.name);
        const codeHtmlPath = path.join(folderPath, 'code.html');

        try {
            // Check if code.html exists
            await fs.access(codeHtmlPath);
            
            const htmlContent = await fs.readFile(codeHtmlPath, 'utf-8');
            
            // Extract title
            const titleMatch = htmlContent.match(/<title>([^<]+)<\/title>/i);
            const title = titleMatch ? titleMatch[1].trim() : entry.name.replace(/_/g, ' ').toUpperCase();
            
            // Check if screen.png exists
            const screenPngPath = path.join(folderPath, 'screen.png');
            let hasScreenshot = false;
            try {
                await fs.access(screenPngPath);
                hasScreenshot = true;
            } catch (e) {}

            const theme = getTheme(htmlContent);
            const category = getCategory(entry.name);

            screens.push({
                folderName: entry.name,
                title: title,
                path: `${entry.name}/code.html`,
                screenshot: hasScreenshot ? `${entry.name}/screen.png` : null,
                category: category,
                theme: theme
            });
        } catch (error) {
            // Skip folders that don't have code.html
        }
    }

    // Sort screens alphabetically by category then by title
    screens.sort((a, b) => {
        if (a.category !== b.category) {
            return a.category.localeCompare(b.category);
        }
        return a.title.localeCompare(b.title);
    });

    const outputPath = path.join(rootDir, 'screens.json');
    await fs.writeFile(outputPath, JSON.stringify(screens, null, 2), 'utf-8');
    console.log(`Successfully indexed ${screens.length} screens. Output saved to: ${outputPath}`);
}

main().catch(console.error);
